### DISCUSSION TITLE
Not working on multilingual websites

### DISCUSSION THREAD

**Discussion by Misagh90** [livewire/livewire#6287] [7. Bugs (archived)]
### Livewire version

V3 beta 8 & V2

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

Hi,
I create a middleware for my multilingual website after updating v2 to v3 all my components not working and goes 404 but the normal routes & blade is OK!
my middleware:
```
class Language
{
    public function handle(Request $request, Closure $next): Response
    {
        if (!in_array($request->segment(1),config('translatable.locales'))) {
            $segments = $request->segments();
            // Set the default language code as the first segment e.g. /en, /fr
            $segments = \Arr::prepend($segments, config('app.fallback_locale'));
            // Redirect to the correct url
            return redirect()->to(implode('/', $segments));
        }
        return $next($request);
    }
}
```

my e.g. routes:

```
Route::group(['prefix' => LaravelLocalization::setLocale(),'middleware' => [ 'localize','localeSessionRedirect', 'localizationRedirect', 'localeViewPath' ]], function () {

    Route::get('/', [App\Http\Controllers\Front\HomeController::class, 'main'])->name('home');
    Auth::routes();
});
```

"http://127.0.0.1:8000/{lang}/livewire/update" show 404 (page not found) error
how can I fix this?

### Please confirm (incomplete submissions will not be addressed)

- [x] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by Misagh90** [comment]
is there any solution?

**Comment by guilebc** [comment]
I have the same problem, there was a solution for version 2 but I can't find anything for version 3.
I saw that the route in the framework was written in hard copy, maybe there's a way to make it customizable?

**Reply by Misagh90** [reply]
In LiveWire V2 you can edit the registerRoutes function in LivewireServiceProvider manually, like this:
`$local = app()->getLocale();` & add $local before the` '/'.$local.'/livewire/upload-file'`

**Comment by PterodactylStore** [comment]
put this route
```php
Livewire::setUpdateRoute(function ($handle) {
    return Route::post('/livewire/update', $handle);
});
```
under localized routes 

https://livewire.laravel.com/docs/upgrading#localization


but the error will happen again when switch betwen pagination or load component with lazy

**Reply by Misagh90** [reply]
the error of uploading files exists on paginations & lazy?!
